cpp/include/raft_runtime/solver/lanczos.hpp

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright (c) 2024-2024, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <raft/core/resources.hpp>
+#include <raft/random/rng_state.hpp>
+
+#include <cstdint>
+
+namespace raft::runtime::solver {
+
+/**
+ * @defgroup rmat_runtime RMAT Runtime API
+ * @{
+ */
+
+#define FUNC_DECL(IndexType, ValueType)              \
+  void lanczos_solver(const raft::resources& handle, \
```

Comment:
Please don't add public APIs that accept raw pointers. These APIs are all deprecated in RAFT. Please accept device_matrix_view, etc.. instead. That's why they exist.

Reply:
Take a look at the other pylibraft cython files for examples of how to do this. pylibraft cython is newer than cuml's cython so we've adopted better design haibits.